### DISCUSSION TITLE
Why am I having trouble setting environment variables?

### DISCUSSION THREAD

**Discussion by MoltenMonster** [fish-shell/fish-shell#10058] [Q&A] [answered]
```
cat '~/.config/fish/config.fish'

set -g STARSHIP_CONFIG "~/.config/fish/starship.toml"
set -g RIPGREP_CONFIG_PATH "~/.config/fish/.ripgreprc"
starship init fish | source⏎
```

But when I do either `env | rg "RIPGREP_CONFIG_PATH"` or `"STARSHIP_CONFIG"` it shows nothing.

**Comment by faho** [comment, marked as answer]
https://fishshell.com/docs/current/faq.html#how-do-i-set-or-clear-an-environment-variable

You have not exported the variables, meaning they are not visible outside of fish, to an external process like, say, starship. Add a `-x` to your set:

```fish
set -gx STARSHIP_CONFIG "~/.config/fish/starship.toml"
set -gx RIPGREP_CONFIG_PATH "~/.config/fish/.ripgreprc"
```
